object/slicer/options.go

```diff
@@ -55,15 +58,29 @@ func (x *Options) SetCopiesNumber(copiesNumber uint32) {
 	x.copiesNumber = copiesNumber
 }
 
-// SetPayloadBuffer sets pre-allocated payloadBuffer to be used to object uploading.
-// For better performance payloadBuffer length should be MaxObjectSize from NeoFS.
+// SetPayloadBuffer sets pre-allocated payloadBuffer to be used to object
+// uploading. The payloadBuffer should have length at least
+// [Options.ObjectPayloadLimit] + 1. If [Options.SetPayloadSize] is called, its
+// argument is used as buffer size requirement instead. Any smaller buffer is
+// ignored.
 func (x *Options) SetPayloadBuffer(payloadBuffer []byte) {
 	x.payloadBuffer = payloadBuffer
 }
 
+// SetPayloadSize allows to specify object's payload size known in advance. if
+// the size is fixed, the option is recommended as it improves the performance
+// of the application using the [Slicer].
+func (x *Options) SetPayloadSize(size uint64) {
+	x.payloadSizeKnown = true
+	x.payloadSize = size
+}
+
 // ObjectPayloadLimit returns required max object size.
 func (x *Options) ObjectPayloadLimit() uint64 {
-	return x.objectPayloadLimit
+	if x.objectPayloadLimit > 0 {
```

Comment:
why this change? i doubt it is expected that simple value holder `Options` with `Set...` and `Get...` may change smth inside. we had the helper func before, what was wrong with it?

Reply:
gone